CONTRIBUTING.md

```diff
@@ -16,6 +16,40 @@ For those just getting started, consult this  [guide](https://help.github.com/ar
 - Do not create methods with more than two parameters (in extremely rare occasions maybe three) except constructors and factory functions. Structs and interfaces exist for a reason.
 - Avoid returning responses (e.g., reconcile.Result, admission.Patched) in anything but Reconcile or Handle functions.
 
+### Cuddling of statements
+Statements must be cuddled, i.e., written as a single block, if an `if`-statement directly follows a single assignment and the condition is directly related to the assignment.
+This commonly occurs with error handling, but is not restricted to it.  
+Example:
+```
+err := assignment1()
```

Comment:
I'd actually even prefer
```suggestion
if err := assignment1(); err != {
}
```
where applicable.

Reply:
Personally I always feel like that makes conditions harder to read. Especially if a function name is longer, but even with shorter ones I feel like that is the case. Having the assignment be separated from the condition makes it clearer what is happening in the condition.

Going a bit more into extreme cases, if the assignment is merged with the condition and the condition has more parts than just checking the error it becomes even more unwieldy. Here an example from the production code:
```
err = dtpullsecret.
		NewReconciler(ctx, controller.client, controller.apiReader, controller.scheme, dynakube, tokens).
		Reconcile()
	if err != nil {
```
That would be an extreme case, but merged it would look like so:
```
if err := dtpullsecret.NewReconciler(ctx, controller.client, controller.apiReader, controller.scheme, dynakube, tokens).Reconcile(); err != nil {
```